datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -821,9 +821,7 @@ impl ExecutionPlan for RepartitionExec {
         child_pushdown_result: ChildPushdownResult,
         _config: &ConfigOptions,
     ) -> Result<FilterPushdownPropagation<Arc<dyn ExecutionPlan>>> {
-        Ok(FilterPushdownPropagation::transparent(
-            child_pushdown_result,
-        ))
+        Ok(FilterPushdownPropagation::all(child_pushdown_result))
```

Comment:
Note that `all` is much more explicit and clear than `transparent`. We could even name it something more verbose?

Reply:
Maybe `if_all`?